sessionctx/stmtctx/stmtctx.go

```diff
@@ -60,6 +60,7 @@ type StatementContext struct {
 	UseCache               bool
 	PadCharToFullLength    bool
 	BatchCheck             bool
+	InNullRejectCheck      bool
```

Comment:
If we make it a parameter of `foldConstant`, we have to modify signature of `FoldConstant`, `NewFunction`, `NewFunctionInternal`, `EvaluateExprWithNull`, this would lead to a lot of unnecessary code changes